compute_horde_sdk/src/compute_horde_core/executor_class.py

```diff
@@ -1,4 +1,9 @@
-from enum import StrEnum
+import sys
+
+if sys.version_info >= (3, 11):  # noqa: UP036
+    from enum import StrEnum
+else:
+    from backports.strenum import StrEnum  # noqa: UP035
```

Comment:
No strong opinions here, but maybe you can follow `easier to ask for forgiveness than permission` rule and use try/except block here (and in other places that are using `sys.version_info`)?
```suggestion
try:
    from enum import StrEnum
except ImportError:
    from backports.strenum import StrEnum
```